bd_spot_wrapper/spot_wrapper/spot.py

```diff
@@ -863,6 +892,21 @@ def convert_transformation_from_BD_to_magnum(self, bd_transformation_dict: dict)
         mn_transformation = mn.Matrix4.from_(rotation_matrix, translation)
         return mn_transformation
 
+    def get_spot_a_T_b(self, a: str, b: str, tree=None) -> mn.Matrix4:
```

Comment:
Could you please also add a doc-string here?

Reply:
And I think this function is important! Do you mind adding line by line comment so that we know each line is doing?

Reply:
This function was copied from Kavit's branch but I have added comments explaining how it works